### PR TITLE
[Merged by Bors] - feat(Algebra/Algebra/Defs): add missing coe lemmas for algebraMap

### PR DESCRIPTION
**Pull request by Command-Master**
Add lemmas `algebraMap.coe_sub`, `algebraMap.coe_natCast` and `algebraMap.coe_smul`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Command-Master** [comment]
I don't like the placement of `coe_smul` in the file, but it uses `algebraMap_eq_smul_one` so it has to be after it

File: Mathlib/Algebra/Algebra/Defs.lean [hunk 1]
```diff
@@ -410,4 +419,11 @@ end Semiring
 
 end Algebra
 
+@[norm_cast]
+theorem algebraMap.coe_smul (A B C : Type*) [SMul A B] [CommSemiring B] [Semiring C] [Algebra B C]
+    [SMul A C] [IsScalarTower A B C] (a : A) (b : B) : (a • b : B) = a • (b : C) := calc
+  ((a • b : B) : C) = (a • b) • 1 := Algebra.algebraMap_eq_smul_one _
+  _ = a • (b • 1) := smul_assoc ..
+  _ = a • (b : C) := congrArg _ (Algebra.algebraMap_eq_smul_one b).symm
```
Comment (eric-wieser) [inline Mathlib/Algebra/Algebra/Defs.lean:427]: I think you can write this as `map_smul (Algebra.ofId B C) a b` or similar?

Comment (Command-Master) [reply Mathlib/Algebra/Algebra/Defs.lean:427]: `Algebra.restrictScalars` is defined in `Algebra/Algebra/Tower`, which comes after `Algebra/Algebra/Defs`

**Comment by j-loreaux** [comment]
looks reasonable to me, but I'd like someone else to look too.

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
Let's try again

bors merge

**Pull request closed without merging**